### ISSUE TITLE
Provide Dial(net, addr) and DialTimeout(network, address string, timeout time.Duration)

### ISSUE THREAD

**Issue by tj** [tinylib/synapse#22]
over the millisecond int?

**Comment by philhofer** [comment]
Yeah, I agree that `DialTimeout("tcp", ":7070", 500*time.Millisecond)` reads better.

The reason there isn't a default `Dial()` (w/o timeout) is because the range of desirable real-world timeouts is pretty enormous. Do you have an opinion about what a reasonable default would be?

**Comment by tj** [comment]
True! A few seconds seems like it would be generous but I agree it might be better to just force people to make their own choice

**Comment by philhofer** [comment]
So I changed `Dial` to use time.Duration. I'm going to leave the issue open in case someone feels strongly about the simpler constructor.

**Comment by tj** [comment]
rad!
